objects/src/accounts/code.rs

```diff
@@ -1,50 +1,67 @@
-use super::{AccountError, Digest, Word};
-use assembly::ModuleAst;
-use crypto::merkle::MerkleTree;
-use miden_core::Program; // TODO: we should be able to import it from the assembly crate
+use super::{AccountError, Digest};
+use assembly::{Assembler, AssemblyContext, LibraryPath, Module, ModuleAst};
+use crypto::merkle::SimpleSmt;
+use miden_lib::MidenLib;
+use miden_stdlib::StdLibrary;
 
 // ACCOUNT CODE
 // ================================================================================================
 
+// CONSTANTS
+// -----------------------------------------------------------------------------------------------
+
+/// The depth of the Merkle tree that is used to commit to the account's public interface.
+const ACCOUNT_CODE_TREE_DEPTH: u8 = 8;
+
 /// Describes public interface of an account.
 ///
 /// Account's public interface consists of a set of account methods, each method being a Miden VM
 /// program. Thus, MAST root of each method commits to the underlying program. We commit to the
 /// entire account interface by building a simple Merkle tree out of all method MAST roots.
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone)] // TODO: Add PartialEq, Eq
 pub struct AccountCode {
-    method_tree: MerkleTree,
+    method_tree: SimpleSmt,
     module: ModuleAst,
-    // methods: Vec<Program>, commented out because Program doesn't currently implement Eq and
-    // PartialEq. Also, there might be a better way of describing a set of programs as they
-    // might share a lot of common code blocks. In a way, we want something like a Program
-    // struct but with many entry points.
+    procedures: Vec<Digest>,
```

Comment:
We should probably settle on either using `method` or `procedure` and use it throughout. I'm fine with either.

Reply:
I think procedure is best to be consistent with terminology used in the `Assembler`.